peer/brontide.go

```diff
@@ -677,7 +670,35 @@ func (p *Brontide) Start() error {
 
 		return lastSerializedBlockHeader[:]
 	}
-	go p.pingHandler(newPingPayload)
+	// TODO(roasbeef): make dynamic in order to
+	// create fake cover traffic
+	// NOTE(proofofkeags): this was changed to be
+	// dynamic to allow better pong identification,
+	// however, more thought is needed to make this
+	// actually usable as a traffic decoy
+	randPongSize := func() uint16 {
+		return uint16(
+			// We don't need cryptographic randomness here.
+			/* #nosec */
+			rand.Intn(lnwire.MaxPongBytes + 1),
+		)
+	}
+
+	pingManager, pingChan, pongChan, failChain := NewPingManager(
+		newPingPayload, randPongSize, pingInterval, pingTimeout)
```

Comment:
> Then rather call pingManger.Stop() from the Brontide Disconnect method rather than passing in the quit channel. The ping manager should have its own quit channel instead.

Adjusted this.

> construction should happen in the Brontide constructor.

While I understand that the pattern throughout the codebase is this, I want to point out the difficulty with doing that in this case.

On the input side we allocate a couple of stack variables to help with some optimizations. We can drop the optimizations, or we will have to put those stack variables into the Brontide struct and I'm reticent to further bloat the Brontide. Dropping the optimizations I'm much more amenable to.

On the output side we would need to also store the ping, pong, and fail channels on the brontide and my goal with the "receive only" and "send only" type casts here is to enlist the go type system to help make sure that these things don't get misused. It is also why I made the pingManagerLiason and the readHandler take these channels as arguments as well, to prevent those values from being made freely available to other goroutines.

While I'm sympathetic to following patterns to make code more uniform, I worry about progressively piling more and more state into the Brontide, which in this context is serving as a global namespace and I don't think it's that controversial to say that large global namespaces are bad for reasoning about code.

How do you think about this tradeoff?

Reply:
> On the output side we would need to also store the ping, pong, and fail channels on the brontide and my goal with the "receive only" and "send only" t

Did you take a look at the patch I sent yesterday?